pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -42,6 +43,40 @@ func (r *Reconciler) appendEnvFrom(envList []corev1.EnvFromSource, secret string
 	})
 }
 
+// appendVolumeMounts append volume mounts pointing to volumes created using secret
+func (r *Reconciler) appendVolumeMounts(vmList []corev1.VolumeMount, volumeName string) []corev1.VolumeMount {
+	for _, vm := range vmList {
+		if vm.Name == volumeName {
+			// volume name already referenced
+			return vmList
+		}
+	}
+
+	return append(vmList, corev1.VolumeMount{
+		Name:      volumeName,
+		MountPath: "/" + volumeName,
```

Comment:
yes

Reply:
Hmm.. not exactly, it will be `/serviceBindingCRName-volumemount`

Reply:
@baijum  

May I suggest we improve the path prefix a bit ?

Let's add a an API attribute in `spec` to let the user specify the base mount path, because this is an information that the user of the ServiceBindingRequest can specify best.
 
```
apiVersion: apps.openshift.io/v1alpha1
kind: ServiceBindingRequest
metadata:
  name: binding-request
  namespace: shoubhik
spec:
  baseVolumePath:"/var/data"
  applicationSelector:
    matchLabels:
      connects-to: postgres
      environment: demo
    resourceKind: DeploymentConfig
  backingServiceSelector:
    resourceKind: postgresql.baiju.dev
    resourceName: Database
    resourceRef: db-demo
    resourceVersion: v1alpha1
```

Because the user has specified `baseVolumePath:"/var/redhat/credentials"`, any binding-driven volume paths would be 
- /var/redhat/credentials/username
- /var/redhat/credentials/password

If the user doesn't specify anything for `baseVolumePath`, then go for a default like
- /var/data/username
- /var/data/password . ( if there are any conventions on how a default data path should look like, go for that. Else, start with `/var/data/*` as a default.

Reply:
Done. ff7be37a8cceb8949dc2fecfa4d06cdfcaa05160